processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -349,10 +368,60 @@ else if ( !method.isUpdateMethod() ) {
                 beforeMappingMethods,
                 afterMappingMethods,
                 finalizeMethod,
-                mappingReferences
+                mappingReferences,
+                subClasses
             );
         }
 
+        private boolean doesNotHaveSubClassMappingsAndCanBeConstructed(Type returnTypeImpl) {
+            return !hasSubclassMappings()
+                && canReturnTypeBeConstructed( returnTypeImpl );
+        }
+
+        private boolean hasSubClassMappingsAndIsEitherAbstractOrCanBeConstructed(Type returnTypeImpl) {
+            return hasSubclassMappings()
+                && isReturnTypeAbstractOrcanBeConstructed( returnTypeImpl );
+        }
+
+        private SubClassMapping createSubClassMapping(SubClassMappingOptions subClassMappingOptions) {
+            TypeFactory typeFactory = ctx.getTypeFactory();
+            Type sourceType = typeFactory.getType( subClassMappingOptions.getSourceClass() );
+            Type targetType = typeFactory.getType( subClassMappingOptions.getTargetClass() );
+
+            SourceRHS rightHandSide = new SourceRHS(
+                "subClassMapping",
+                sourceType,
+                Collections.emptySet(),
+                "SubClassMapping for " + sourceType.getFullyQualifiedName() );
+            SelectionCriteria criteria =
+                SelectionCriteria
+                                 .forMappingMethods(
+                                     new SelectionParameters(
+                                         Collections.emptyList(),
+                                         Collections.emptyList(),
+                                         subClassMappingOptions.getTargetClass(),
+                                         ctx.getTypeUtils() ).withSourceRHS( rightHandSide ),
+                                     null,
+                                     null,
+                                     false );
+            Assignment assignment = ctx
+                                   .getMappingResolver()
+                                   .getTargetAssignment(
+                                       method,
+                                       null,
+                                       targetType,
+                                       FormattingParameters.EMPTY,
+                                       criteria,
+                                       rightHandSide,
+                                       null,
+                                           () -> forgeMapping( rightHandSide, sourceType, targetType ) );
```

Comment:
I actually think that this invalidates the statement that a method is needed for the `SubClassMapping`.

Can we also have some tests where the configuration from the base mapping is passed to the forged mappings as well?

Reply:
This was a good idea, found a bug in the code with it.
Apparently the ForgedMethod that I created didn't get the MappingReferences of the parent method.
Fixing that as well now.